setup.py

```diff
@@ -87,15 +91,29 @@ def reduce_requirements(key):
 setup_requires = []
 install_requires = requirements['install']
 
-cupy_pkg = None
-try:
-    cupy_pkg = pkg_resources.get_distribution('cupy')
-except pkg_resources.DistributionNotFound:
-    pass
 
+def find_any_distribution(pkgs):
+    for pkg in pkgs:
+        try:
+            return pkg_resources.get_distribution(pkg)
+        except pkg_resources.DistributionNotFound:
+            pass
+    return None
+
+
+cupy_pkg = find_any_distribution([
```

Comment:
Do you think we should add a comment here for why we simply pick the first available distribution?

Reply:
Thanks, added comment.